spec/contracts/placeholder_users/shared_contract_examples.rb

```diff
@@ -33,41 +33,50 @@
 shared_examples_for 'placeholder user contract' do
   let(:placeholder_user_name) { 'UX Designer' }
 
-  it_behaves_like 'contract is valid for active admin users only'
+  context 'when user with global permission' do
+    let(:current_user) { FactoryBot.create(:user, global_permission: %i[add_placeholder_user]) }
 
-  context 'name' do
-    context 'is valid' do
-      it_behaves_like 'contract is valid'
-    end
+    it_behaves_like 'contract is valid'
+  end
 
-    context 'is not too long' do
-      let(:placeholder_user) { PlaceholderUser.new(name: 'X' * 257) }
+  it_behaves_like 'contract is valid for active admin users only'
```

Comment:
This sentence looks wrong to me. If a non-admin own the global right to add and edit placeholder users, then it would also be valid.

Reply:
The shared example doesn't test any global permission, it just tests active admin(valid) + inactive admins(invalid) + regular users(invalid)

Reply:
Maybe 'contract is valid for active admins and invalid for regular users'?